src/client/index.js

```diff
@@ -0,0 +1,267 @@
+const sig = require("@tendermint/sig");
+const _ = require("lodash");
+const tx = require("../tx").tx;
+const msg = require("../msg").msg;
+const utils = require("../utils").utils;
+
+const prefix = "kava";
+const derivationPath = "m/44'/459'/0'/0/0";
+
+const api = {
+  nodeInfo: "/node_info",
+  getSwap: "bep3/swap",
+  getSwaps: "/bep3/swaps",
+  getCDP: "cdp/cdp",
+  getCDPs: "/cdp/cdps",
+  getAuction: "/auction/auction",
+  getAuctions: "/auction/auctions"
+};
+
+/**
+ * The Kava client.
+ */
+class KavaClient {
+  /**
+   * @param {String} server Kava public url
+   */
+  constructor(server) {
+    if (!server) {
+      throw new Error("Kava server should not be null");
+    }
+    this.baseURI = server;
+  }
+
+  /**
+   * Initialize the client with the chain's ID. Asynchronous.
+   * @return {Promise}
+   */
+  async initChain() {
+    if (!this.chainID) {
+      const res = await tx.getTx(api.nodeInfo, this.baseURI);
+      this.chainID = _.get(res, "data.node_info.network");
+    }
+    return this;
+  }
+
+  /**
+   * Manually set the chain's ID
+   * @param {String} chainID Kava chain ID
+   */
+  setChainID(chainID) {
+    if (!chainID) {
+      throw new Error("chainID cannot be undefined");
+    }
+    this.chainID = chainID;
+    return this;
+  }
+
+  /**
+   * Manually set the wallet's account number
+   * @param {String} accNum Account number of the Kava address
+   */
+  setAccountNumber(accNum) {
+    if (!accNum) {
+      throw new Error("account number cannot be undefined");
+    }
+    this.accNum = String(accNum);
+    return this;
+  }
+
+  /**
+   * Set the client's wallet which is used for signature generation
+   * @param {String} mnemonic Kava address mnemonic
+   * @return {Promise}
+   */
+  setWallet(mnemonic) {
+    if (!mnemonic) {
+      throw new Error("mnemonic cannot be undefined");
+    }
+    this.wallet = sig.createWalletFromMnemonic(
+      mnemonic,
+      "",
+      prefix,
+      derivationPath
+    );
+    return this;
+  }
```

Comment:
If the other wallet args are passed as optional args with defaults then you can still just call `setWallet("some mnemonic")` in most cases, but more advanced users can pass in the other bits as needed.

```suggestion
  setWallet(mnemonic, derivationPath=derivationPath , password="", prefix=prefix) {
    if (!mnemonic) {
      throw new Error("mnemonic cannot be undefined");
    }
    this.wallet = sig.createWalletFromMnemonic(
      mnemonic,
      "",
      prefix,
      derivationPath
    );
    return this;
  }
```

Reply:
Should the full derivation path be parameterized or just the coin ID?

Reply:
What do you think about adding ledger support? Not within the scope of this PR, but making an issue for it and adding in the future.

Reply:
https://github.com/Kava-Labs/javascript-sdk/pull/1/commits/787d9274b3095e334872e9f012e3856595323cf4